core/src/main/java/io/questdb/griffin/engine/functions/table/ReadParquetPageFrameCursor.java

```diff
@@ -0,0 +1,272 @@
+    public SymbolTable newSymbolTable(int columnIndex) {
+        return null;
+    }
+
+    @Override
+    public @Nullable PageFrame next() {
+        final int rowGroupIndex = ++frame.rowGroupIndex;
+        if (rowGroupIndex < rowGroupCount) {
+            frame.rowGroupSize = decoder.metadata().rowGroupSize(rowGroupIndex);
+            frame.partitionLo = frame.partitionHi;
+            frame.partitionHi = frame.partitionHi + frame.rowGroupSize;
+            return frame;
+        }
+        return null;
+    }
+
+    public void of(LPSZ path) {
+        // Reopen the file, it could have changed
+        this.fd = TableUtils.openRO(ff, path, LOG);
+        this.fileSize = ff.length(fd);
+        this.addr = TableUtils.mapRO(ff, fd, fileSize, MemoryTag.MMAP_PARQUET_PARTITION_DECODER);
+        decoder.of(addr, fileSize, MemoryTag.NATIVE_PARQUET_PARTITION_DECODER);
+        if (metadataHasChanged(metadata, decoder)) {
+            // We need to recompile the factory as the Parquet metadata has changed.
+            throw TableReferenceOutOfDateException.of(path);
+        }
+        // TODO(puzpuzpuz): we need to support projection pushdown for read_parquet()
+        //                  as currently we're decoding all columns
+        columnIndexes.clear();
+        for (int i = 0, n = metadata.getColumnCount(); i < n; i++) {
+            columnIndexes.add(i);
+        }
+        this.rowCount = decoder.metadata().rowCount();
+        this.rowGroupCount = decoder.metadata().rowGroupCount();
+        toTop();
+    }
+
+    @Override
+    public long size() throws DataUnavailableException {
+        return rowCount;
+    }
+
+    @Override
+    public boolean supportsSizeCalculation() {
+        return true;
+    }
+
+    @Override
+    public void toTop() {
+        frame.clear();
+    }
+
+    private boolean metadataHasChanged(RecordMetadata metadata, PartitionDecoder decoder) {
+        if (metadata.getColumnCount() != decoder.metadata().columnCount()) {
+            return true;
+        }
+        for (int i = 0, n = metadata.getColumnCount(); i < n; i++) {
+            final int metadataType = metadata.getColumnType(i);
+            final int decoderType = decoder.metadata().getColumnType(i);
+
+            boolean remappingDetected = symbolToVarcharRemappingDetected(metadataType, decoderType);
+            if (remappingDetected) {
+                continue;
+            }
+            if (metadata.getColumnType(i) != decoder.metadata().getColumnType(i)) {
+                return true;
+            }
+        }
+        for (int i = 0, n = metadata.getColumnCount(); i < n; i++) {
+            if (!Chars.equals(metadata.getColumnName(i), decoder.metadata().columnName(i))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean symbolToVarcharRemappingDetected(int metadataType, int decoderType) {
+        return metadataType == ColumnType.VARCHAR && decoderType == ColumnType.SYMBOL;
```

Comment:
It would be easire for me to understand the code if this method was inlined. The name is more confusing than the condition it checks.

Reply:
Inlined in 91ebac0a0c4cc4970f72e9aeef0ed65952ee71b8